src/zeta/aca_zeta_programming.cpp

```diff
@@ -49,15 +49,15 @@ int ACA_Zeta_Programming::create_or_update_zeta_config(const alcor::schema::AuxG
               not_care_culminative_time);
     }
   }
-  uint32_t oam_server_port = current_AuxGateway.zeta_info().port_inband_operation();
+  uint oam_server_port = current_AuxGateway.zeta_info().port_inband_operation();
 
-  uint32_t oam_port = ACA_Vlan_Manager::get_instance().get_oam_server_port(vpc_id);
+  uint oam_port = ACA_Vlan_Manager::get_instance().get_oam_server_port(tunnel_id);
   // oam_server_port is not set
   if (oam_port == 0) {
-    ACA_Vlan_Manager::get_instance().set_oam_server_port(vpc_id, oam_server_port);
+    ACA_Vlan_Manager::get_instance().set_oam_server_port(tunnel_id, oam_server_port);
 
     //update oam_ports_table and add the OAM punt rule also if this is the first port in the VPC
-    Aca_Oam_Port_Manager::get_instance().add_vpc(oam_server_port, tunnel_id);
+    Aca_Oam_Port_Manager::get_instance().add_port_number(oam_server_port);
   }
   // add the group bucket rule
   overall_rc = _create_or_update_zeta_group_entry(&stZetaCfg);
```

Comment:
line 63 is adding the group rules for:
1. first port in the VPC - fine
2. second port in same VPC - also adding the group rule when it is not needed
3. first port in a different VPC which uses the same ZGC - also adding the group rule when it is not needed

We need to optimize the code so that only the first port using the ZGC will add the group rule